client/src/api/api.ts

```diff
@@ -194,20 +193,206 @@ export interface Anime {
   ending_themes: string[];
 }
 
-export const categorisedResultsResponse = async (category: string, query: string) => {
-  const results = await axios.get(`http://localhost:8080/search/${category}/${query}`);
+export interface AnimeVideos {
+  promo: [
+    {
+      title: string;
+      image_url: string;
+      video_url: string;
+    }
+  ];
+  episodes: [
+    {
+      title: string;
+      episode: string;
+      url: string;
+      image_url: string;
+    }
+  ];
+}
+
+export interface AnimeCharacters {
+  characters: [
+    {
+      mal_id: number;
+      image_url: string;
+      name: string;
+      role: string;
+      voice_actors: [
+        {
+          mal_id: number;
+          name: string;
+          imague_url: string;
+          language: string;
+        }
+      ];
+    }
+  ];
+}
+
+export interface AnimeReviews {
+  reviews: [
+    {
+      mal_id: number;
+      content: string;
+      date: string;
+      reviewer: {
+        username: string;
+        scores: {
+          overall: number;
+        };
+      };
+    }
+  ];
+}
+
+export interface AnimeNews {
+  articles: [
+    {
+      title: string;
+      date: string;
+      author_name: string;
+      image_url: string;
+      intro: string;
+    }
+  ];
+}
+
+export interface AnimeRecommendations {
+  recommendations: [
+    {
+      mal_id: number;
+      image_url: string;
+      title: string;
+      recommendation_count: number;
+    }
+  ];
+}
+
+export interface CombinedAnimeResponse {
+  details: Anime;
+  videos: AnimeVideos;
+  characters: AnimeCharacters;
+  reviews: AnimeReviews;
+  news: AnimeNews;
+  recommendations: AnimeRecommendations;
+}
+
+export const categorisedResultsResponse = async (
+  category: string,
+  query: string
+) => {
+  const results = await axios.get(
+    `http://localhost:8080/search/${category}/${query}`
+  );
+
+  return results.data;
+};
+
+export const topResultsResponse = async (
+  category: string,
+  subtype?: string
+) => {
+  const results = await axios.get(
+    `http://localhost:8080/top/${category}/1/${subtype ?? ""}`
+  );
+
+  return results.data;
+};
+
+export const animeDetailsResponse = async (
+  category: string,
+  id: number
+): Promise<Anime> => {
+  const results = await axios.get(`http://localhost:8080/${category}/${id}/`);
+
+  return results.data;
+};
+
+export const animeVideosResponse = async (
+  category: string,
+  id: number
+): Promise<AnimeVideos> => {
+  const results = await axios.get(
+    `http://localhost:8080/${category}/${id}/videos`
+  );
+
+  return results.data;
+};
+
+export const animeCharactersResponse = async (
+  category: string,
+  id: number
+): Promise<AnimeCharacters> => {
+  const results = await axios.get(
+    `http://localhost:8080/${category}/${id}/characters_staff`
+  );
 
   return results.data;
```

Comment:
if you don't want to also get the staff members from this request, you should be able to go `return results.data.characters`